### ISSUE TITLE
Look into CockroachDB as a storage backend

### ISSUE THREAD

**Issue by dominikh** [tracer/tracer#17]
CockroachDB is a distributed SQL database. It uses the postgres wire protocol and recently added support for joins. It may be possible to adapt our postgres store with no or minimal changes. This would allow for an even easier deployment, and eliminate one of the major downsides of the postgres backend: difficulty distribution of data.

**Comment by dmitshur** [comment]
Question, have you considered (and are you aware of) the BoltDB database? I know it's much lower level than a SQL database like Postgres, so perhaps that makes it not viable.

**Comment by dominikh** [comment]
Yes, BoltDB had been considered. It's an embedded key/value store, meaning it's not distributed and we'd need to build that on top of it, and essentially build our own distribtued database system. There is an existing one, but it didn't meet some requirements, either.
